src/ardupilot/ArduPilotProtocol.h

```diff
@@ -0,0 +1,111 @@
+#include "../network/websocket/WebSocketProtocol.h"
+#include "../network/websocket/WebSocketServer.h"
+#include "../world_interface/data.h"
+
+#include <mutex>
+
+namespace net {
+namespace ardupilot {
+
+/* @brief A protocol in order to handle messages from an ArduPilot enabled device */
+class ArduPilotProtocol {
+public:
+	ArduPilotProtocol();
+	~ArduPilotProtocol();
+	ArduPilotProtocol(const ArduPilotProtocol& other) = delete;
+	ArduPilotProtocol& operator=(const ArduPilotProtocol& other) = delete;
+
+	/* @brief Gets the last reported GPS position (lat, lon)
+	 *
+	 * @return Last reported GPS position
+	 */
+	robot::types::DataPoint<navtypes::gpscoords_t> getGPS();
+
+	/* @brief Gets the last reported IMU readings (roll, pitch, yaw)
+	 *
+	 * @return Last reported IMU readings
+	 */
+	robot::types::DataPoint<navtypes::eulerangles_t> getIMU();
+
+	/* @brief Gets the last reported heading (degrees)
+	 *
+	 * @return Last reported heading
+	 */
+	robot::types::DataPoint<int> getHeading();
+
+	/* @brief Validates GPS request
+	 *
+	 * @param json message
+	 *
+	 * @return True if message is valid, false otherwise
+	 */
+	static bool validateGPSRequest(const nlohmann::json& j);
```

Comment:
Should all these validate and handle methods be public? Seems like we can make em private